.gitattributes

```diff
@@ -0,0 +1,2 @@
+# Normalize line endings to LF regardless of OS or git settings
+* text=auto
```

Comment:
I'm a little uneasy about this as is because I've seen this break binary assets due to git incorrectly detecting them as text and forcing LFs on the end of them in other repos many times (e.g. for images and such).  Debugging that is a nightmare because it often isn't clear that it's broken until some time later and it's unclear why because binary asserts are unreadable by diffs.

There are two options: either be explicit about the file extensions it applies to or add explicit overrides for the binary files in the repo.  Perhaps the latter would be easier and acceptable since text content is **much** more common than binary content.

Taking that approach, I think the following would work:

```
# Normalize line endings of text files to LF regardless of OS or git settings
*            text=auto
*.bz2        -text
*.hex        -text
```

Reply:
I was also a little nervous about using a wildcard to select every filetype. There's not a ton of different filetypes; maybe it would be better to take option 1 you suggested instead. Especially because I see a `.gz` file, and seven files without a specific filetype.

```Powershell
> Get-ChildItem -Recurse -File | Select-Object -ExpandProperty Extension | Group-Object | Sort-Object Count -Descending | Format-Table Name, Count

Name           Count
----           -----
.go              666
.md               84
.mod              29
.sum              22
.hex              15
                   7
.bz2               7
.py                7
.json              5
.sh                5
.mediawiki         2
.yml               2
.conf              2
.gitattributes     1
.gv                1
.dockerignore      1
.svg               1
.service           1
.xml               1
.gitignore         1
.gz                1
```

Going with option 1 seems less potentially dangerous? And if someone ends up trying to push a CRLF change to a new filetype, it's not that hard to just update this file.

What about:

```
# Normalize line endings of text files to LF regardless of OS or git settings
*.conf          text=auto
*.dockerignore  text=auto
.gitattributes  text=auto
.gitignore      text=auto
*.go            text=auto
*.gv            text=auto
*.json          text=auto
*.md            text=auto
*.mediawiki     text=auto
*.mod           text=auto
*.py            text=auto
*.service       text=auto
*.sh            text=auto
*.sum           text=auto
*.svg           text=auto
*.xml           text=auto
*.yml           text=auto
```

Alphabetized? Sorted another way?

Reply:
I do prefer option 1 overall since it's more explicit and can't break by default.  I wasn't totally opposed to option 2 though since I know that git generally does a pretty good job of binary vs text detection, so it's usually fine except for a few cases like images and archives at times.

At any rate, your proposed updated looks good except it should just be `text` when specifying extensions like that because they are known to be text.  The `text=auto` case instructs `git` to do the aforementioned binary detection which is why it's used with `*` since it matches everything and thus needs to decide.

From https://git-scm.com/docs/gitattributes:

> When text is set to "auto", Git decides by itself whether the file is text or binary. If it is text and the file was not already in Git with CRLF endings, line endings are converted on checkin and checkout as described above. Otherwise, no conversion is done on checkin or checkout.